src/CommunityToolkit.Maui.Core/Views/Popup/PopupExtensions.android.cs

```diff
@@ -109,7 +110,30 @@ public static void SetSize(this Dialog dialog, in IPopup popup, in AView contain
 
 		CalculateSizes(popup, context, ref realWidth, ref realHeight, ref realContentWidth, ref realContentHeight);
 
-		var childLayoutParams = (FrameLayout.LayoutParams)(child.LayoutParameters ?? throw new NullReferenceException());
+		if (context.GetWindow() is IWindow windowManager)
+		{
+			var resourceIdStatus = context.Resources?.GetIdentifier("status_bar_height", "dimen", "android") ?? 0;
+			var resourceIdNavigation = context.Resources?.GetIdentifier("navigation_bar_height", "dimen", "android") ?? 0;
```

Comment:
Is it possible to accomplish the same results without searching `Resources` for an identifier? 

I see that this is the recommended approach in a few StackOverflow answers, but I'd prefer to avoid using hard-coded strings to search `Resources` because it could easily become a breaking change in a future version of Android or .NET MAUI. 

Could you confirm that this is the only way to accomplish it before we merge this in?

Reply:
@brminnick, Thank you for your review.
I understand that the hardcoded code below is not the desired code.

    var resourceIdStatus = context.Resources?.GetIdentifier("status_bar_height", "dimen", "android") ?? 0;
    var resourceIdNavigation = context.Resources?.GetIdentifier("navigation_bar_height", "dimen", "android") ?? 0;

Therefore, I also think that a method that does not use these is preferable.

If you check the source code below on the .NET MAUI side, methods are defined to get the 
height of the StatusBar and NavigationBar. They are GetStatusBarHeight, GetNavigationBarHeight methods.

[src\Platform\Android\ContextExtensions.cs]

An implementation similar to the one above exists below and is used to get the height of 
the status bar on the NavigationPageRender class.

[src\Compatibility\Core\src\Android\AppCompat\NavigationPageRenderer.cs]

It is also possible to write using these as follows.

    if (context.GetWindow() is IWindow windowManager)
    {
        var statusBarHeight = context.GetStatusBarHeight();
        var navigationBarHeight = context.GetNavigationBarHeight();
        realWidth = realWidth <= windowManager.Width * density ? realWidth : (int)(windowManager.Width * density - (windowManager.Height < windowManager.Width ? navigationBarHeight : 0));
        realHeight = realHeight <= (windowManager.Height * density - ((windowManager.Height < windowManager.Width ? 0 : navigationBarHeight) + statusBarHeight)) ? realHeight : (int)(windowManager.Height * density - ((windowManager.Height < windowManager.Width ? 0 : navigationBarHeight) + statusBarHeight));
        window.SetLayout(realWidth, realHeight);
    }

However, there are pitfalls. Checking the implementation of the GetStatusBarHeight and GetNavigationBarHeight methods, they are written in the hard-coded way I posted the other day. I just delegated the implementation to 
the .NET MAUI side.

So these are not the preferred methods.

It can also be written as follows.

    Android.Graphics.Rect? contentRect = new Android.Graphics.Rect();
    decorView.GetWindowVisibleDisplayFrame(contentRect);
    int contentWidth = contentRect?.Width() ?? throw new NullReferenceException();
    int contentHeight = contentRect?.Height() ?? throw new NullReferenceException();
    realWidth = realWidth <= contentWidth ? realWidth : contentWidth;
    realHeight = realHeight <= contentHeight ? realHeight : contentHeight;
    window.SetLayout(realWidth, realHeight);

Although it is a simple way of writing that does not depend on the API level, it also has pitfalls.
If you try to display the popup while the software keyboard is displayed, the height will 
be obtained without the height of the software keyboard.

It can be avoided by forcibly closing the software keyboard, but if there is an input field 
in the Popup and the focus is assigned, the Popup will be resized.

Although it is necessary to distinguish cases for each API level, it is possible to write without 
hardcoding by writing as follows.

    var windowManager = window.WindowManager;
    if (windowManager is not null)
    {
        int windowWidth;
        int windowHeight;
        int statusBarHeight;
        int navigationBarHeight;

        if (OperatingSystem.IsAndroidVersionAtLeast((int)BuildVersionCodes.R))
        {
            WindowMetrics windowMetrics = windowManager.CurrentWindowMetrics;
            AInsets insets = windowMetrics.WindowInsets.GetInsetsIgnoringVisibility(WindowInsets.Type.SystemBars());
            windowWidth = windowMetrics.Bounds.Width();
            windowHeight = windowMetrics.Bounds.Height();
            statusBarHeight = insets.Top;
            navigationBarHeight = windowHeight < windowWidth ? insets.Left + insets.Right : insets.Bottom;
        }
        else 
        {
            APoint? realSize = new APoint();
            APoint? displaySize = new APoint();
            ARect? contentRect = new ARect();
            windowManager.DefaultDisplay?.GetRealSize(realSize);
            windowManager.DefaultDisplay?.GetSize(displaySize);
            decorView.GetWindowVisibleDisplayFrame(contentRect);
            windowWidth = realSize?.X ?? throw new InvalidOperationException($"{nameof(realSize.X)} cannot be null");
            windowHeight = realSize?.Y ?? throw new InvalidOperationException($"{nameof(realSize.Y)} cannot be null");
            int displayWidth = displaySize?.X ?? throw new InvalidOperationException($"{nameof(displaySize.X)} cannot be null");
            int displayHeight = displaySize?.Y ?? throw new InvalidOperationException($"{nameof(displaySize.Y)} cannot be null");
            statusBarHeight = contentRect.Top;
            navigationBarHeight = windowHeight < windowWidth ? (windowWidth - displayWidth) : (windowHeight - displayHeight);
        }

        realWidth = realWidth <= (windowWidth - (windowHeight < windowWidth ? navigationBarHeight : 0)) ? realWidth : (int)(windowWidth - (windowHeight < windowWidth ? navigationBarHeight : 0));
        realHeight = realHeight <= (windowHeight - ((windowHeight < windowWidth ? 0 : navigationBarHeight) + statusBarHeight)) ? realHeight : (int)(windowHeight - ((windowHeight < windowWidth ? 0 : navigationBarHeight) + statusBarHeight));
        window.SetLayout(realWidth, realHeight);
    }

Versions before API level 30 use decorView, but only to get the height of the status bar.
As mentioned above, when the software keyboard is displayed, the height of the window is the 
height excluding the height of the software keyboard.

Since the display position of the navigation bar changes depending on whether the screen 
is vertical or horizontal, it is taken into consideration.

Reply:
Thanks for doing this! The updates look good.